File: src/test/java/com/josephbaca/world/RoomTest.java
package com.josephbaca.world;

import com.josephbaca.rpggame.Game;
import com.josephbaca.util.LoggerUtilKt;
import io.vavr.collection.List;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RoomTest {

  private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(RoomTest.class);

  @Test
  void testRoomMovement() {
//    LoggerUtilKt.setLogLevel("DEBUG", "root");

    List<String> commands = List.of("up", "right", "left", "down");

    List<List<Boolean>> success = commands.map(
        command -> List.fill(10, () -> false).map(item -> {
          Game g = new Game();

          // Ensure not at edge of world
          g.input("up");
          g.input("right");

          String description1 = g.input("where");
          String description2 = g.input(command);

          return !description1.equals(description2);
        })
    );

    for (List<Boolean> llb : success) {
      LOG.info(llb.toString());
      assertTrue(llb.fold(false, (l, r) -> l || r));
    }
  }

}

File: src/test/java/com/josephbaca/world/RoomTest.kt
package com.josephbaca.world

import com.josephbaca.rpggame.GameWrapper
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

internal class RoomTest {

    @Test
    fun roomMovementChangesRoom() {

        val commands = listOf("forward", "back", "left", "right").map { "go $it" }

        commands.map { command ->
            val game = GameWrapper()
            val contextManager = game.gameStateManager

            // Ensure not at edge of world
            game.getGameReponseFromInput("go forward")
            game.getGameReponseFromInput("go right")

            val room1 = contextManager.currentContext
            game.getGameReponseFromInput(command)
            val room2 = contextManager.currentContext

            room1 != room2
        }.forEach { assertTrue(it) }

    }

    @Test
    fun roomMovementChangesOrientation() {

        val commands = listOf("back", "left", "right").map { "go $it" }

        commands.map { command ->
            val game = GameWrapper()
            val contextManager = game.gameStateManager

            // Ensure not at edge of world
            game.getGameReponseFromInput("go forward")
            game.getGameReponseFromInput("go right")

            val orientation1 = contextManager.world.playerOrientation
            game.getGameReponseFromInput(command)
            val orientation2 = contextManager.world.playerOrientation

            orientation1 != orientation2
        }.forEach { assertTrue(it) }

    }

    @Test
    fun roomMovementDoesntChangeOrientation() {

        val commands = listOf("forward").map { "go $it" }

        commands.map { command ->
            val game = GameWrapper()
            val contextManager = game.gameStateManager

            // Ensure not at edge of world
            game.getGameReponseFromInput("go forward")
            game.getGameReponseFromInput("go right")

            val orientation1 = contextManager.world.playerOrientation
            game.getGameReponseFromInput(command)
            val orientation2 = contextManager.world.playerOrientation

            orientation1 == orientation2
        }.forEach { assertTrue(it) }

    }

}